### PR TITLE
Solar panels

### PR DESCRIPTION
**Pull request by pexel5**
## About The Pull Request

Fixing solar panels and things for them.

## Why It's Good For The Game

Other way to get power, (if techno fuck up the SM). With solar panels you can power up your department or all ship, if you want it.

## Changelog
:cl:
add: Add boards recipes to circuits disk and solar assembly to parts disk
fix: fixed solar control console, solar panels and solar tracker

/:cl:

### CODE REVIEW

**Comment by pexel5** [comment]
we fix it boys, WE FIX CHANGELOGS

File: code/datums/autolathe/circuits.dm [hunk 1]
```diff
@@ -109,3 +109,7 @@
 /datum/design/autolathe/circuit/autolathe_excelsior
 	name = "excelsior autoforge"
 	build_path = /obj/item/electronics/circuitboard/excelsiorautolathe
+
+/datum/design/autolathe/circuit/tracker
```
Comment (SirRichardFrancis) [inline code/datums/autolathe/circuits.dm:113]: Under Excelsior circuits? Why?

File: code/modules/power/solar.dm [hunk 2]
```diff
@@ -81,7 +81,8 @@
 			broken()
 		else
 			new /obj/item/material/shard(src.loc)
-			new /obj/item/material/shard(src.loc)
+			new /obj/item/stack/rods(src.loc)
+			new /obj/item/stack/rods(src.loc)
```
Comment (SirRichardFrancis) [inline code/modules/power/solar.dm:85]:
```suggestion
			new /obj/item/stack/rods(loc)
			new /obj/item/stack/rods(loc)
```

File: code/modules/power/solar.dm [hunk 3]
```diff
@@ -203,6 +204,7 @@
 	w_class = ITEM_SIZE_BULKY // Pretty big!
 	anchored = FALSE
 	price_tag = 100
+	matter = list(MATERIAL_STEEL = 5, MATERIAL_PLASTIC = 5, MATERIAL_PLASTEEL = 2,MATERIAL_SILVER = 1)
```
Comment (SirRichardFrancis) [inline code/modules/power/solar.dm:207]:
```suggestion
	matter = list(MATERIAL_STEEL = 5, MATERIAL_PLASTIC = 5, MATERIAL_PLASTEEL = 2, MATERIAL_SILVER = 1)
```

File: code/modules/power/solar.dm [hunk 4]
```diff
@@ -245,27 +247,34 @@
...
-			var/obj/item/stack/material/S = I
-			if(S.use(2))
-				glass_type = I.type
-				playsound(src.loc, 'sound/machines/click.ogg', 50, 1)
-				user.visible_message(SPAN_NOTICE("[user] places the glass on the solar assembly."))
-				if(tracker)
-					new /obj/machinery/power/tracker(get_turf(src), src)
-				else
-					new /obj/machinery/power/solar(get_turf(src), src)
+	if(istype(I, /obj/item/stack/material) && (I.get_material_name() == MATERIAL_GLASS || I.get_material_name() == MATERIAL_RGLASS))
```
Comment (SirRichardFrancis) [inline code/modules/power/solar.dm:250]:
```suggestion
	if(istype(I, /obj/item/stack/material/glass))
```
Think that should be enough.

File: code/modules/power/solar.dm [hunk 5]
```diff
@@ -245,27 +247,34 @@
...
+		if(!anchored)
+			to_chat(user, span_warning("You need to secure the assembly before you can add glass."))
+			return
+		var/turf/solarturf = get_turf(src)
+		if(locate(/obj/machinery/power/solar) in solarturf)
```
Comment (SirRichardFrancis) [inline code/modules/power/solar.dm:255]:
```suggestion
		if(locate(/obj/machinery/power/solar) in get_turf(src))
```
Muh speed.

File: code/modules/power/solar.dm [hunk 6]
```diff
@@ -245,27 +247,34 @@
...
+			to_chat(user, span_warning("A solar panel is already assembled here."))
+			return
+		var/obj/item/stack/material/S = I
+		if(S.use(2))
+			glass_type = S.type
+			playsound(src.loc, 'sound/machines/click.ogg', 50, TRUE)
```
Comment (SirRichardFrancis) [inline code/modules/power/solar.dm:261]:
```suggestion
			playsound(loc, 'sound/machines/click.ogg', 50, TRUE)
```

File: code/modules/power/solar.dm [hunk 7]
```diff
@@ -245,27 +247,34 @@
...
+			to_chat(user, span_warning("You need two sheets of glass to put them into a solar panel!"))
 			return
+		return TRUE
 
 	if(!tracker)
 		if(istype(I, /obj/item/electronics/tracker))
 			tracker = 1
 			user.drop_item()
 			qdel(I)
+			playsound(src.loc, 'sound/machines/click.ogg', 50, TRUE)
```
Comment (SirRichardFrancis) [inline code/modules/power/solar.dm:277]:
```suggestion
			playsound(loc, 'sound/machines/click.ogg', 50, TRUE)
```

**Review by SirRichardFrancis — CHANGES_REQUESTED** [review]

File: code/modules/power/solar.dm [hunk 8]
```diff
@@ -245,27 +247,33 @@
...
+			playsound(loc, 'sound/machines/click.ogg', 50, TRUE)
+			user.visible_message(span_notice("[user] places the glass on the solar assembly."), span_notice("You place the glass on the solar assembly."))
+			if(tracker)
+				new /obj/machinery/power/tracker(get_turf(src), src)
 			else
-				to_chat(user, SPAN_WARNING("You need two sheets of glass to put them into a solar panel."))
-				return
+				new /obj/machinery/power/solar(get_turf(src), src)
+		else
+			to_chat(user, span_warning("You need two sheets of glass to put them into a solar panel!"))
```
Comment (Firefox13) [inline code/modules/power/solar.dm:267]:
```suggestion
			to_chat(user, SPAWN_WARNING("You need two sheets of glass to put them into a solar panel!"))
```

File: code/modules/power/solar.dm [hunk 9]
```diff
@@ -245,27 +247,33 @@
...
+			to_chat(user, span_warning("You need to secure the assembly before you can add glass."))
+			return
+		if(locate(/obj/machinery/power/solar) in get_turf(src))
+			to_chat(user, span_warning("A solar panel is already assembled here."))
+			return
+		var/obj/item/stack/material/S = I
+		if(S.use(2))
+			glass_type = S.type
+			playsound(loc, 'sound/machines/click.ogg', 50, TRUE)
+			user.visible_message(span_notice("[user] places the glass on the solar assembly."), span_notice("You place the glass on the solar assembly."))
```
Comment (Firefox13) [inline code/modules/power/solar.dm:261]:
```suggestion
			user.visible_message(SPAN_NOTICE("[user] places the glass on the solar assembly."), SPAN_NOTICE("You place the glass on the solar assembly."))
```

File: code/modules/power/solar.dm [hunk 10]
```diff
@@ -245,27 +247,33 @@
...
-				if(tracker)
-					new /obj/machinery/power/tracker(get_turf(src), src)
-				else
-					new /obj/machinery/power/solar(get_turf(src), src)
+	if(istype(I, /obj/item/stack/material/glass))
+		if(!anchored)
+			to_chat(user, span_warning("You need to secure the assembly before you can add glass."))
+			return
+		if(locate(/obj/machinery/power/solar) in get_turf(src))
+			to_chat(user, span_warning("A solar panel is already assembled here."))
```
Comment (Firefox13) [inline code/modules/power/solar.dm:255]:
```suggestion
			to_chat(user, SPAN_WARNING("A solar panel is already assembled here."))
```

File: code/modules/power/solar.dm [hunk 11]
```diff
@@ -245,27 +247,33 @@
...
-				glass_type = I.type
-				playsound(src.loc, 'sound/machines/click.ogg', 50, 1)
-				user.visible_message(SPAN_NOTICE("[user] places the glass on the solar assembly."))
-				if(tracker)
-					new /obj/machinery/power/tracker(get_turf(src), src)
-				else
-					new /obj/machinery/power/solar(get_turf(src), src)
+	if(istype(I, /obj/item/stack/material/glass))
+		if(!anchored)
+			to_chat(user, span_warning("You need to secure the assembly before you can add glass."))
```
Comment (Firefox13) [inline code/modules/power/solar.dm:252]:
```suggestion
			to_chat(user, SPAN_WARNING("You need to secure the assembly before you can add glass."))
```

File: code/modules/power/solar.dm [hunk 12]
```diff
@@ -203,6 +204,7 @@
 	w_class = ITEM_SIZE_BULKY // Pretty big!
 	anchored = FALSE
 	price_tag = 100
+	matter = list(MATERIAL_STEEL = 5, MATERIAL_PLASTIC = 5, MATERIAL_PLASTEEL = 2, MATERIAL_SILVER = 1)
```
Comment (Firefox13) [inline code/modules/power/solar.dm:207]:
```suggestion
	matter = list(MATERIAL_STEEL = 5, MATERIAL_PLASTIC = 5, MATERIAL_SILVER = 1)
```
Plasteel is probably a bit too much.
There should be a way to get solar panels outside of autolathe - like guild. Or technomancers could spawn with one of those "Going green!" crates.

Comment (Humonitarian) [reply code/modules/power/solar.dm]: > Solar panels are made of tempered glass, which is sometimes called toughened glass.

I would suggest replacing plasteel with 5 glass instead, maybe reducing the steel/plastic price too

Comment (pexel5) [reply code/modules/power/solar.dm]: dont forgot, this just 1 panel, you need a lot for generate.

Comment (Firefox13) [reply code/modules/power/solar.dm]: Sheets are inserted in a separate step...

**Review by SirRichardFrancis — APPROVED** [review]

File: code/modules/power/solar.dm (hunk 12 above)
Comment (pexel5) [reply code/modules/power/solar.dm]: > Sheets are inserted in a separate step...

he mean, replace plasteel in recipe on glass. Btw accept PR please )

**Review by Pink-Chink — APPROVED** [review]

**Comment by Jan-Blasiak** [comment]
Do you plan to add shield interaction in the future?

**Comment by pexel5** [comment]
> Do you plan to add shield interaction in the future?

Most likely not, my skills are still not enough

**Pull request merged**